### ISSUE TITLE
Error fix for Slim 2.3.1 with Twig ~1

### ISSUE THREAD

**Issue by sutonium** [slimphp/Slim#590]
Here's my composer.json

```
"require": {
        "slim/slim": "~2",
        "slim/extras": "~2",
        "twig/twig": "~1",
        "propel/propel1": "~1"
    }
```

I have error after upgrading from 2.2.0 to 2.3.1

```
$app->get('/', function () use ($app) {
    $app->render('index.html');
});
```

```
Details
Type: ErrorException
Code: 4096
Message: Argument 1 passed to Twig_Template::render() must be an array, object given, called in /var/www/big-mobile/vendor/slim/extras/Slim/Extras/Views/Twig.php on line 99 and defined
File: /var/www/big-mobile/vendor/twig/twig/lib/Twig/Template.php
Line: 244
```

I fix that by changing file: **_vendor/slim/extras/Slim/Extras/Views/Twig.php line: 99**_ from

```
return $template->render($this->data);
```

into

```
return $template->render($this->data->get('data', array()));
```

Now the error's gone.

**Comment by sutonium** [comment]
Another fix is using package `"slim/views": "0.1.*"`

and the code example perhaps like this

```
$twig = new \Slim\Views\Twig();
$twig->parserOptions = array(
    'charset' => 'utf-8',
    'cache' => realpath('../templates/cache'),
    'auto_reload' => true,
    'strict_variables' => false,
    'autoescape' => true
);

$twig->parserExtensions = array(
    new \Slim\Views\TwigExtension(),
);

$app = new \Slim\Slim(array(
    'templates.path' => '../templates',
    'log.level' => 4,
    'log.enabled' => true,
    'log.writer' => new \Slim\Extras\Log\DateTimeFileWriter(array(
        'path' => '../logs',
        'name_format' => 'y-m-d'
    )),
    'view' => $twig
));

$app->get('/', function () use ($app) {
    $app->render('index.html');
});
```

**Comment by codeguy** [comment]
You likely need to use `$view->all()` to fetch an array of its data to pass into the template. This is resolved if you use the latest versions of Slim and Slim-Views.
